src/notebooks/controllers/installPythonControllerCommands.ts

```diff
@@ -62,6 +74,18 @@ export class InstallPythonControllerCommands implements IExtensionSingleActivati
         );
     }
 
+    // Track if there are any cells currently executing or pending
+    private onDidChangeNotebookCellExecutionState(stateEvent: NotebookCellExecutionStateChangeEvent) {
+        if (
+            stateEvent.state === NotebookCellExecutionState.Pending ||
+            stateEvent.state === NotebookCellExecutionState.Executing
+        ) {
```

Comment:
Might want to  check the notebook type and ensure we only look at Jupyter & IW notebooks.